src/actions.js

```diff
@@ -250,15 +250,28 @@ export function login(credentials) {
               refreshExpiresIn
             }
           }`;
-      await dispatch(
-        graphqlMutation(mutation, credentials, ["CORE_AUTH_LOGIN_REQ", "CORE_AUTH_LOGIN_RESP", "CORE_AUTH_ERR"]),
-      );
+      try {
+        // Dispatch GraphQL mutation and handle response
+        const response = await dispatch(
+          graphqlMutation(mutation, credentials, ["CORE_AUTH_LOGIN_REQ", "CORE_AUTH_LOGIN_RESP", "CORE_AUTH_ERR"]),
+        );
+        if (response.payload && response.payload.errors && response.payload.errors.length > 0) {
+          const errorMessage = response.payload.errors[0].message;
+          dispatch(authError({ message: errorMessage }));
+          return { loginStatus: "CORE_AUTH_ERR", message: errorMessage };
+        }
+        const action = await dispatch(loadUser());
+        return { loginStatus: action.type, message: action?.payload?.response?.detail ?? "" };
+      } catch (error) {
+        dispatch(authError({ message: error.message }));
+        return { loginStatus: "CORE_AUTH_ERR", message: error.message };
+      }
     } else {
       // Try to refresh the token using the cookie (if present)
```

Comment:
If possible, remove unnecessary comments from the component. Comments should only be used to explain specific details.